client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -1586,4 +1708,134 @@ private boolean connectFail(String message) {
   public TransportClientFactory getDataClientFactory() {
     return dataClientFactory;
   }
+
+  class ReviveRequest {
+    public int mapId;
+    public int attemptId;
+    public int partitionId;
+    public PartitionLocation loc;
+    public StatusCode cause;
+    public volatile StatusCode reviveStatus;
+
+    public ReviveRequest(
+        int mapId, int attemptId, int partitionId, PartitionLocation loc, StatusCode cause) {
+      this.mapId = mapId;
+      this.attemptId = attemptId;
+      this.partitionId = partitionId;
+      this.loc = loc;
+      this.cause = cause;
+      reviveStatus = StatusCode.REVIVE_INITIALIZED;
+    }
+  }
+
+  class ReviveManager {
+    // appid -> shuffleId -> requests
+    HashMap<String, HashMap<Integer, Set<ReviveRequest>>> pendingRequests = new HashMap<>();
+    HashMap<String, HashMap<Integer, Set<ReviveRequest>>> inProcessRequests = null;
+    private ScheduledExecutorService batchReviveRequestScheduler =
+        ThreadUtils.newDaemonSingleThreadScheduledExecutor("worker-forward-message-scheduler");
+
+    public ReviveManager() {
+      batchReviveRequestScheduler.scheduleAtFixedRate(
+          new Runnable() {
+            @Override
+            public void run() {
+              // We must ensure the swap is atomic
+              synchronized (this) {
+                inProcessRequests = pendingRequests;
+                pendingRequests = new HashMap<>();
+              }
+              if (!inProcessRequests.isEmpty()) {
+                for (Map.Entry<String, HashMap<Integer, Set<ReviveRequest>>> entry :
+                    inProcessRequests.entrySet()) {
+                  String appId = entry.getKey();
+                  HashMap<Integer, Set<ReviveRequest>> shuffleMap = entry.getValue();
+                  for (Map.Entry<Integer, Set<ReviveRequest>> shuffleEntry :
+                      shuffleMap.entrySet()) {
+                    // Call reviveBatch for requests in the same (appId, shuffleId)
+                    int shuffleId = shuffleEntry.getKey();
+                    Set<ReviveRequest> requests = shuffleEntry.getValue();
+                    Map<Integer, Integer> partitionMaxEpoch = new HashMap<>();
+                    Set<Integer> mapIds = new HashSet<>();
+                    ArrayList<ReviveRequest> filteredRequests = new ArrayList<>();
```

Comment:
Use List<ReviveRequest> to keep the unifiled code style? (Maybe somewhere has same problem).